src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -0,0 +1,283 @@
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Test for Create Opportunities from Recurring Donations
+*
+*/
+@isTest(IsParallel=true)
+private with sharing class RD2_OpportunityService_TEST {
+
+    private static final String AFFILIATION_OCR_ROLE_SOLICITOR = 'Solicitor';
+    private static final String RELATIONSHIP_TYPE_EMPLOYEE = 'Employee';
+    private static final String RELATIONSHIP_STATUS_CURRENT = 'Current';
+    private static final String PLEDGED_STAGENAME = 'Pledged';
+    private static final String PROMISED_STAGENAME = 'Promised';
+    private static final String CREDITCARD = 'Credit Card';
+
+    /***
+    * @description Verifies if it creates opportunities when an recurring donation
+      is inserted with a contact and the stage is pledged.
+    */
+    @isTest
+    private static void shouldCreateOppWhenRDIsInsertedWithContact() {
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<npe03__Recurring_Donation__c,List<Date>> rdsWithCloseDates = new Map<npe03__Recurring_Donation__c,List<Date>>();
+        List<Date> CloseDate = new List<Date> ();
+        List<npe03__Recurring_Donation__c> rds = new List<npe03__Recurring_Donation__c> ();
+        List<npe03__Recurring_Donation__c> recurringDonations = new List<npe03__Recurring_Donation__c> ();
+        List<Opportunity> oppCreated = new List<Opportunity> ();
+        List<npe01__OppPayment__c> paymentForOpps = new List<npe01__OppPayment__c> ();
+        String oppNameFormat = '';
+
+        CloseDate.add(System.today()+2);
+        CloseDate.add(System.today()+5);
+
+        createRdSettingsWithStage(PLEDGED_STAGENAME);
+
+        List<Contact> primaryContacts = UTIL_UnitTestData_TEST.createMultipleTestContacts(1);
+        insert primaryContacts;
+        primaryContacts = getContacts(primaryContacts);
+
+        Campaign campaign = new Campaign(Name = 'TeaCampaign', IsActive = true);
+        insert campaign;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
```

Comment:
These should use the builder method on the class to construct the right format RD
```suggestion
        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
```